modules/agar/src/main/ts/ephox/agar/api/Pipeline.ts

```diff
@@ -31,10 +31,6 @@ const runStep = <T, U> (initial: T, step: Step<T, U>, onSuccess: NextFn<U>, onFa
   step.runStep(initial, onSuccess, onFailure, TestLogs.getOrInit(initLogs));
 };
 
-/**
- * @deprecated Use runStep instead
- * TODO: Remove
- */
```

Comment:
See comment on runStep:

> If you need to run a sequence of steps, compose them using the functions in StepSequence

So, the reason I did this is that `async` takes a `Step<any, any>[]` and loses the type information between steps. `Step.compose` overcomes this. So, the suggestion is to compose the steps using `StepSequence.compose` or `StepSequence.composeSame`.

Reply:
That's not a viable replacement given this is used in every test and bloats boilerplate code, which we don't want to do.

Is there any reason we can't just make `Pipeline.async` use the above?

Reply:
It's all about the type information. 

If we don't care about the type information, sure, use StepSequence.sequenceUnsafe. 

If we do care about the type information, use StepSequence.sequence, but you'd need Pipeline.async to have the crazy type. Which is doable. I think we do care about the type information because there are types in there that we want; there are casts we don't want, and we can make mistakes in our tests without them.

Another thought would be to change our pattern slightly. e.g. instead of 

```
UnitTest.testAsync('label', (success, failure) => {
  Pipeline.async({}, [
    step1,
    step2
  ], () => success(), failure);
}
```

we could do something like:

```
stepTest('label', () => {

  return StepSequence.compose([
    step1,
    step2
  ]);
});
```

Reply:
Also, I can't shake the feeling that this would all be better for TypeScript if we used async..await, instead.

Reply:
James wanted to spike out a version of all this that used async/await, but I don't believe he ever got to it (see TINY-4620).

As for the rest of it, that's a discussion for another day or something better suited for slack than here. For now, we discussed this on Friday and Pipeline.async should not be deprecated as it's part of the boilerplate code.

Reply:
👍  cool

Reply:
I should add, I'd definitely be in favour of using async/await for this sort of thing. It'd save us having to duplicate core code into Steps, etc... and would likely feel more natural to new devs.